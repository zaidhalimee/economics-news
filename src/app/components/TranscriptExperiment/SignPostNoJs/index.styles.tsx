import pixelsToRem from '#app/utilities/pixelsToRem';
import { css, Theme } from '@emotion/react';

const WIDTH = 250;

const styles = {
  container: ({ palette, spacings, mq }: Theme) =>
    css({
      textAlign: 'center',
      position: 'absolute',
      color: palette.WHITE,
      padding: `${spacings.FULL}rem`,
      display: 'flex',
      flexWrap: 'nowrap',
      background: `rgba(34, 34, 34, 0.75)`,
      height: '100%',
      width: '100%',
      zIndex: '1',
      [mq.GROUP_1_MIN_WIDTH]: {
        textAlign: 'end',
        left: `calc(100% - ${pixelsToRem(WIDTH)}rem)`,
        width: `${pixelsToRem(WIDTH)}rem`,
        height: 'unset',
      },
    }),
  message: ({ palette }: Theme) =>
    css({
      alignItems: 'center',
      display: 'flex',
      color: palette.WHITE,
    }),
};

export default styles;
