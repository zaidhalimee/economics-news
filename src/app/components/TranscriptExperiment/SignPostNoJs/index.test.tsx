import React from 'react';
import { render } from '../../react-testing-library-with-providers';
import SignPostNoJs from '.';

describe('Sign Post', () => {
  afterEach(() => {
    jest.resetModules();
  });

  it(`should render a 'Help reduce your power consumption' message`, () => {
    const { container } = render(
      <SignPostNoJs noJsMessage="Help reduce your power and data usage by not playing video content." />,
      {
        service: 'news',
      },
    );
    expect(container).toMatchSnapshot();
  });
});
