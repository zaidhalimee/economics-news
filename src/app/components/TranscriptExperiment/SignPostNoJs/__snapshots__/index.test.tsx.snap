// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Post should render a 'Help reduce your power consumption' message 1`] = `
<div>
  <noscript />
</div>
`;
